### ISSUE TITLE
Build Error on snort 3.1.70 with openwrt platform 

### ISSUE THREAD

**Issue by yidianyipie** [snort3/snort3#359]
Hi Team, 

I build snort3 on openwrt platform and it shows error which confused me very much

```python
make[3]: Entering directory '/home/runner/work/AutoBuild_Lede/AutoBuild_Lede/openwrt/feeds/packages/net/snort3'
rm -f /home/runner/work/AutoBuild_Lede/AutoBuild_Lede/openwrt/build_dir/target-x86_64_musl/snort3-3.1.70.0/.built
touch /home/runner/work/AutoBuild_Lede/AutoBuild_Lede/openwrt/build_dir/target-x86_64_musl/snort3-3.1.70.0/.built_check
MAKEFLAGS="" /home/runner/work/AutoBuild_Lede/AutoBuild_Lede/openwrt/staging_dir/host/bin/ninja  -j1 -C /home/runner/work/AutoBuild_Lede/AutoBuild_Lede/openwrt/build_dir/target-x86_64_musl/snort3-3.1.70.0 
ninja: Entering directory `/home/runner/work/AutoBuild_Lede/AutoBuild_Lede/openwrt/build_dir/target-x86_64_musl/snort3-3.1.70.0'
[1/818] Building CXX object src/main/CMakeFiles/main.dir/oops_handler.cc.o
FAILED: src/main/CMakeFiles/main.dir/oops_handler.cc.o 
/home/runner/work/AutoBuild_Lede/AutoBuild_Lede/openwrt/staging_dir/toolchain-x86_64_gcc-11.3.0_musl/bin/x86_64-openwrt-linux-musl-g++ -DHAVE_CONFIG_H -Dinline=inline -Drestrict=__restrict -I/home/runner/work/AutoBuild_Lede/AutoBuild_Lede/openwrt/build_dir/target-x86_64_musl/snort3-3.1.70.0/src/network_inspectors -I/home/runner/work/AutoBuild_Lede/AutoBuild_Lede/openwrt/build_dir/target-x86_64_musl/snort3-3.1.70.0/src -I/home/runner/work/AutoBuild_Lede/AutoBuild_Lede/openwrt/staging_dir/target-x86_64_musl/usr/include/luajit-2.1 -I/home/runner/work/AutoBuild_Lede/AutoBuild_Lede/openwrt/build_dir/target-x86_64_musl/snort3-3.1.70.0 -I/home/runner/work/AutoBuild_Lede/AutoBuild_Lede/openwrt/staging_dir/host/include -I/home/runner/work/AutoBuild_Lede/AutoBuild_Lede/openwrt/staging_dir/target-x86_64_musl/usr/include/uuid -I/home/runner/work/AutoBuild_Lede/AutoBuild_Lede/openwrt/build_dir/target-x86_64_musl/snort3-3.1.70.0/src/main -Os -pipe -fno-caller-saves -fno-plt -fhonour-copts -fmacro-prefix-map=/home/runner/wo
make[3]: *** [Makefile:115: /home/runner/work/AutoBuild_Lede/AutoBuild_Lede/openwrt/build_dir/target-x86_64_musl/snort3-3.1.70.0/.built] Error 1
make[2]: *** [package/Makefile:116: package/feeds/packages/snort3/compile] Error 1
/home/runner/work/AutoBuild_Lede/AutoBuild_Lede/openwrt/build_dir/target-x86_64_musl/snort3-3.1.70.0/src/main/oops_handler.cc: In member function 'void OopsHandler::set_current_message(DAQ_Msg_h, snort::SFDAQInstance*)':
/home/runner/work/AutoBuild_Lede/AutoBuild_Lede/openwrt/build_dir/target-x86_64_musl/snort3-3.1.70.0/src/main/oops_handler.cc:61:9: error: 'DIOCTL_GetPrivDataLen' was not declared in this scope
   61 |         DIOCTL_GetPrivDataLen ioctl_data = {cur_msg,  0};
      |         ^~~~~~~~~~~~~~~~~~~~~
/home/runner/work/AutoBuild_Lede/AutoBuild_Lede/openwrt/build_dir/target-x86_64_musl/snort3-3.1.70.0/src/main/oops_handler.cc:62:48: error: 'DIOCTL_GET_PRIV_DATA_LEN' was not declared in this scope
   62 |         if (DAQ_SUCCESS == daq_instance->ioctl(DIOCTL_GET_PRIV_DATA_LEN, &ioctl_data, sizeof(ioctl_data)))
      |                                                ^~~~~~~~~~~~~~~~~~~~~~~~
/home/runner/work/AutoBuild_Lede/AutoBuild_Lede/openwrt/build_dir/target-x86_64_musl/snort3-3.1.70.0/src/main/oops_handler.cc:62:75: error: 'ioctl_data' was not declared in this scope
   62 |         if (DAQ_SUCCESS == daq_instance->ioctl(DIOCTL_GET_PRIV_DATA_LEN, &ioctl_data, sizeof(ioctl_data)))
      |                                                                           ^~~~~~~~~~
/home/runner/work/AutoBuild_Lede/AutoBuild_Lede/openwrt/build_dir/target-x86_64_musl/snort3-3.1.70.0/src/main/oops_handler.cc: In member function 'void OopsHandler::eternalize(int)':
/home/runner/work/AutoBuild_Lede/AutoBuild_Lede/openwrt/build_dir/target-x86_64_musl/snort3-3.1.70.0/src/main/oops_handler.cc:97:27: error: 'daq_msg_get_priv_data' was not declared in this scope; did you mean 'daq_msg_get_data'?
   97 |         memcpy(priv_data, daq_msg_get_priv_data(msg), std::min<size_t>(priv_data_len, sizeof(priv_data)));
      |                           ^~~~~~~~~~~~~~~~~~~~~
      |                           daq_msg_get_data
ninja: build stopped: subcommand failed.
make[3]: Leaving directory '/home/runner/work/AutoBuild_Lede/AutoBuild_Lede/openwrt/feeds/packages/net/snort3'
time: package/feeds/packages/snort3/compile#0.32#0.04#0.35
    ERROR: package/feeds/packages/snort3 failed to build.
make[1]: *** [package/Makefile:110: /home/runner/work/AutoBuild_Lede/AutoBuild_Lede/openwrt/staging_dir/target-x86_64_musl/stamp/.package_compile] Error 2
make[2]: Leaving directory '/home/runner/work/AutoBuild_Lede/AutoBuild_Lede/openwrt'
make[[1](https://github.com/yidianyipie/AutoBuild_Lede/actions/runs/8583768480/job/23523358185#step:15:1)]: Leaving directory '/home/runner/work/AutoBuild_Lede/AutoBuild_Lede/openwrt'
make: *** [/home/runner/work/AutoBuild_Lede/AutoBuild_Lede/openwrt/include/toplevel.mk:231: world] Error 2
Error: Process completed with exit code 2.

```

Could you please give me some advice on this build problem

BR
Robin

**Comment by matirko** [comment]
Looks like you might be missing [this commit](https://github.com/snort3/libdaq/commit/b3f3c2c3fb43ff52ffefd4ea70a14f3b9b5ff05a). If you're using an older libdaq, I'd recommend grabbing a more current version and trying to rebuild.

**Comment by yidianyipie** [comment]
Hi @matirko 

The problem was resolved by change libdaq version from 3.0.11 to 3.0.14 

Thanks for your help 

BR
Robin

**Comment by kimboslice99** [comment]
I too cannot build snort3 for openwrt.

Sure would be nice if the package maintainer would do their job. Bah
